x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/RolesRetrievalResult.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.authz.store;
+
+import org.elasticsearch.xpack.core.security.authz.RoleDescriptor;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public final class RolesRetrievalResult {
+
+    public static final RolesRetrievalResult EMPTY = new RolesRetrievalResult();
+    public static final RolesRetrievalResult SUPERUSER = new RolesRetrievalResult();
```

Comment:
Can we merge the code for this and that `static` block somehow. It bothers me that `EMPTY` and `SUPERUSER` are the same at construction time.

Either:
```
public static final RolesRetrievalResult SUPERUSER;
static {
 SUPERUSER = new RolesRetrievalResult();
 SUPERUSER.addDescriptors(Set.of(ReservedRolesStore.SUPERUSER_ROLE_DESCRIPTOR));
}
```
or

```
     public static final RolesRetrievalResult SUPERUSER = superuserRolesRetrievalResult();
     private static RolesRetrievalResult superuserRolesRetrievalResult() { ... }
```
?
```

Reply:
I took the first approach by merging them in the static block.